torch/csrc/api/include/torch/nn/functional/batchnorm.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <torch/types.h>
+#include <torch/cuda.h>
+
+namespace torch {
+namespace nn {
+namespace functional {
+
+inline Tensor batch_norm(const Tensor& input, const Tensor& running_mean,
+                         const Tensor& running_var, const Tensor& weight,
+                         const Tensor& bias, bool training,
+                         double momentum, double eps) {
+  if (training) {
+    std::vector<int64_t> size = input.sizes().vec();
```

Comment:
I suspect we could do
```suggestion
    auto size = input.sizes();
```
under the hood it uses `ArrayRef` as type for `size` :D

Reply:
Thanks. I followed your instruction.